File: src/test/java/com/znsio/teswiz/screen/dineout/DineoutLandingScreen.java
package com.znsio.teswiz.screen.dineout;


public abstract class DineoutLandingScreen {

    public static DineoutLandingScreen get() {
        return com.znsio.teswiz.screen.ScreenRegistry.getScreen(DineoutLandingScreen.class);
    }

    public abstract DineoutLandingScreen selectDefaultCity();

    public abstract DineoutLandingScreen selectCity(String city);

    public abstract DineoutLandingScreen searchCuisine(String cusine);
}

File: src/test/resources/playwright/screens/dineout/dineout-landing.screen.ts
import type { ScreenContext } from "../screen-context.ts";
import { stayOnCurrentScreen, unsupportedScreenAction } from "../screen-route.ts";

const LOCATORS = {
  defaultCity: '//a[@aria-label="Mumbai"]',
  restaurantSearch: "#restaurantSearch",
  locationFilter: '//li[text()="Location"]',
  submitCity: '//section//button[@value="Submit"]',
} as const;

export async function selectDefaultCity(screen: ScreenContext): Promise<object> {
  await screen.page.locator(`xpath=${LOCATORS.defaultCity}`).click();
  return stayOnCurrentScreen();
}

export async function selectCity(screen: ScreenContext, city: string): Promise<object> {
  const restaurantSearch = screen.page.locator(LOCATORS.restaurantSearch);
  await restaurantSearch.clear();
  await restaurantSearch.fill(city);
  await screen.page.locator(`xpath=${LOCATORS.locationFilter}`).click();
  await screen.page.locator(`xpath=${LOCATORS.submitCity}`).click();
  return stayOnCurrentScreen();
}

export function searchCuisine(): object {
  return unsupportedScreenAction(
    "Dineout cuisine search is not supported on web for WEB_ENGINE=playwright-ts.",
  );
}
